src/nnet3bin/nnet3-show-progress.cc

```diff
@@ -107,17 +107,16 @@ int main(int argc, char *argv[]) {
                                                  eg_end = examples.end();
         for (; eg_iter != eg_end; ++eg_iter)
           prob_computer.Compute(*eg_iter);
-        const SimpleObjectiveInfo *objf_info = prob_computer.GetObjective("output");
-        double objf_per_frame = objf_info->tot_objective / objf_info->tot_weight;
+        double tot_weight = 0.0; // sum of weights for all objectives involved in egs.
+        std::string obj_info = prob_computer.GetAllObjectiveInfo(&tot_weight);
```

Comment:
Please revert the changes you made to this file, and remove the function GetAllObjectiveInfo(), and instead at line 83 of this file, change:
`if (!examples_rspecifier.empty()) {`
to:
`if (!examples_rspecifier.empty() && IsSimpleNnet(nnet1)) {`
The diagnostics that this program prints are not even that important, we don't need to go to any trouble to get them to work in this case, and I'm scared your modified code may break or be unsuitable in some other circumstances.